modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/Metadata.java

```diff
@@ -162,4 +162,10 @@ public void addHeader(String name, String value) {
         headers.write(buffer);
         return bytes;
     }
+
+    public void setBytes(byte[] bytes) throws Exception {
```

Comment:
Can we do this in a constructor (i.e. creation of properties and headers)?

Reply:
fixed!